modular_skyrat/modules/modular_items/lewd_items/code/lewd_items/attachable_vibrator.dm

```diff
@@ -228,27 +228,27 @@
 			soundloop2.start()
 		if(vibration_mode == "high")
 			soundloop3.start()
-
-	if(toy_on == FALSE)
+	else
 		soundloop1.stop()
 		soundloop2.stop()
 		soundloop3.stop()
 
 	//adding messages to chat if vibrator enabled while
-	var/mob/living/carbon/human/U = loc
-	if(toy_on == TRUE)
-		if(src == U.penis || U.vagina || U.anus)
-			to_chat(U, span_purple("You feel pleasant vibrations deep below..."))
-		if(src == U.nipples)
-			to_chat(U, span_purple("You feel pleasant stimulation in your nipples."))
-	if(toy_on == FALSE && (src == U.penis || src == U.vagina || src == U.anus || src == U.nipples))
-		to_chat(U, span_purple("The vibrating toy no longer drives you mad."))
-
-	if(master)
-		if(isassembly(master))
-			var/obj/item/assembly/master_as_assembly = master
-			master_as_assembly.pulsed()
-		master.receive_signal()
+	var/mob/living/carbon/human/vibrated = loc
+	if(toy_on)
+		if(src == vibrated.penis || vibrated.vagina || vibrated.anus)
```

Comment:
will this really work as intended? should it not be:
> if(src == vibrated.penis || src == vibrated.vagina || src == vibrated.anus)

Reply:
good catch